packages/apollo-cli/src/config.ts

```diff
@@ -0,0 +1,267 @@
+import { basename, dirname, join, relative } from "path";
+import { fs, withGlobalFS } from "apollo-codegen-core/lib/localfs";
+
+import * as fg from "glob";
+import * as minimatch from "minimatch";
+import {
+  GraphQLSchema,
+  buildClientSchema,
+  extendSchema,
+  visit,
+  buildASTSchema
+} from "graphql";
+import { loadSchema } from "./load-schema";
+import { loadQueryDocuments } from "apollo-codegen-core/lib/loading";
+
+export interface EndpointConfig {
+  url?: string; // main HTTP endpoint
+  subscriptions?: string; // WS endpoint for subscriptions
+  headers?: Object; // headers to send when performing operations
+}
+
+export interface SchemaDependency {
+  schema?: string;
+  endpoint?: EndpointConfig;
+  engineKey?: string;
+  extends?: string;
```

Comment:
how do we handle multiple extends? I'm assuming right now we don't

Reply:
So for me multiple extends didn't make too much sense because you're taking a schema and adjusting it. I'm not sure what it would mean to extend multiple schemas. In the future though, I would like to support multiple documents extending a schema in the case that you split up your types in multiple files.